<a href="https://colab.research.google.com/github/RaquelFonsec/Teste-Z-valor-P/blob/main/Teste_de_Hipotese.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Uma pesquisa de mercado revelou que 240 dentre 600 pessoas de uma localidade, escolhidas aleatoriamente, comprariam certo produto se o mesmo for lançado no mercado da localidade. Construa intervalos de confiança de 95% e 99% para a proporção de pessoas da localidade que comprariam o referido produto.


In [1]:
import numpy as np
from scipy.stats import norm

# Dados do problema
n = 600  # Tamanho da amostra
x = 240  # Número de pessoas que comprariam
p_hat = x / n  # Proporção amostral

# Valores críticos Z para 95% e 99%
z_95 = norm.ppf(0.975)  # 1.96
z_99 = norm.ppf(0.995)  # 2.576

# Erro padrão da proporção
EP = np.sqrt((p_hat * (1 - p_hat)) / n)

# Intervalo de Confiança 95%
IC_95_min = p_hat - z_95 * EP
IC_95_max = p_hat + z_95 * EP

# Intervalo de Confiança 99%
IC_99_min = p_hat - z_99 * EP
IC_99_max = p_hat + z_99 * EP

# Exibindo os resultados
print(f"Intervalo de Confiança 95%: ({IC_95_min:.4f}, {IC_95_max:.4f})")
print(f"Intervalo de Confiança 99%: ({IC_99_min:.4f}, {IC_99_max:.4f})")


Intervalo de Confiança 95%: (0.3608, 0.4392)
Intervalo de Confiança 99%: (0.3485, 0.4515)


O fabricante de certa marca de suco informa que as embalagens de seu produto têm em média 500 ml, com desvio padrão igual a 10 ml. Tendo sido encontradas no mercado algumas embalagens com menos de 500 ml, suspeita-se que a informação do fabricante seja falsa. Para verificar se isto ocorre, um fiscal analisa uma amostra de 200 embalagens escolhidas aleatoriamente no mercado e constata que as mesmas contêm em média 498 ml. Considerando-se um nível de significância de 5%, pode-se afirmar que o fabricante está correto? Calcule o valor da prova para esta amostra.


In [2]:
from scipy.stats import norm

# Estatística Z calculada
Z = -2.83

# Cálculo do p-valor
p_valor = norm.cdf(Z)
print(f"P-valor: {p_valor:.5f}")


P-valor: 0.00233


Como o p-valor = 0.0023 é menor que o nível de significância de 0.05, rejeitamos a hipótese nula
𝐻
0
H
0
​
 . Isso significa que há evidências estatísticas para afirmar que a média das embalagens é menor do que 500 ml, confirmando a suspeita de que o fabricante está errado. 🚨

📌 Conclusão final: O fabricante não está correto, pois as embalagens contêm menos do que 500 ml em média.

A duração das lâmpadas produzidas por certo fabricante tem distribuição normal com média igual a 1200 horas e desvio padrão igual a 300 horas. O fabricante introduz um novo processo na produção das lâmpadas. Para verificar se o novo processo produz lâmpadas de maior duração, o fabricante observa 100 lâmpadas produzidas pelo novo processo e constata que as mesmas duram em média 1265 horas. Admitindo-se um nível de significância de 5%, pode-se concluir que o novo processo produz lâmpadas com maior duração?

In [3]:
from scipy.stats import norm

# Estatística Z calculada
Z = 2.17

# Cálculo do p-valor (1 - P(Z < 2.17))
p_valor = 1 - norm.cdf(Z)
print(f"P-valor: {p_valor:.5f}")


P-valor: 0.01500


 Conclusão



Como
𝑝
-
𝑣
𝑎
𝑙
𝑜
𝑟
=
0.0150
p-valor=0.0150 é menor que o nível de significância de 0.05, rejeitamos a hipótese nula
𝐻
0
H
0
​
 . Isso significa que há evidências estatísticas para afirmar que a média de duração das lâmpadas aumentou com o novo processo.

📌 Conclusão final: Sim, o novo processo produz lâmpadas com maior duração, pois há evidências estatísticas para confirmar esse aumento. ✅

O custo de produção de certo artigo numa localidade tem distribuição normal com média igual a R$42,00. Desenvolve-se uma política de redução de custos na empresa para melhorar a competitividade do referido produto no mercado. Observando-se os custos de 100 unidades deste produto, obteve-se uma média de 34,9 e um desvio padrão de 3,96. Admitindo-se um nível de significância de 5%, pode-se afirmar que o custo do produto considerado diminuiu?


In [4]:
from scipy.stats import norm

# Estatística Z calculada
Z = -17.93

# Cálculo do p-valor (P(Z < -17.93))
p_valor = norm.cdf(Z)
print(f"P-valor: {p_valor:.10f}")


P-valor: 0.0000000000


Conclusão




Como
𝑝
-
𝑣
𝑎
𝑙
𝑜
𝑟
≈
0
p-valor≈0 é menor que o nível de significância de 0.05, rejeitamos a hipótese nula
𝐻
0
H
0
​
 . Isso significa que há fortes evidências estatísticas para afirmar que o custo médio realmente diminuiu.

📌 Conclusão final: Sim, a política de redução de custos foi eficaz, pois há evidências estatísticas de que o custo médio do produto diminuiu. ✅